chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
-than to have to scroll around repeatedly inside a given file.
-Just like you named each of the analysis datasets,
-each of the individual analysis files should be descriptively named.
-Code files such as \path{spatial-diff-in-diff.do},
-\path{matching-villages.do}, and \path{summary-statistics.do}
-are clear indicators of what each file is doing
-and allow code to be found very quickly.
-Eventually, you will need to compile a ``release package''
-that links the outputs more structurally to exhibits,
-but at this stage function is more important than structure.
-
-Outputs should, whenever possible, be created in lightweight formats.
-For graphics,\sidenote{\url{https://dimewiki.worldbank.org/wiki/Checklist:_Reviewing_Graphs}} \texttt{.eps} or \texttt{.tif} files are acceptable;
-for tables,\sidenote{\url{https://dimewiki.worldbank.org/wiki/Checklist:_Submit_Table}} \texttt{.tex} is preferred (don't worry about making it
-really nicely formatted until you are ready to publish).
-Excel \texttt{.xlsx} files are also acceptable,
-although if you are working on a large report
-they will become cumbersome to update after revisions.
-If you are writing your code in Git/GitHub,
-you should output results into that directory,
-not the directory where data is stored (i.e. Dropbox).
-Results will differ across different branches
-and will constantly be re-writing each other:
-you should have each result stored with the code that created it.
 
-%------------------------------------------------
+% Exploratory and final data analysis -----------------------------------------
+The analysis stage usually starts with a process we call exploratory data analysis.
+This is when you are trying different things and looking for patterns in your data. 
+It progresses into final analysis when your team starts to decide what are the main results, those that will make it into the research output.
+The way you deal with code and outputs for exploratory and final analysis is different, and this section will discuss how.
+% Organizing scripts ---------------------------------------------------------
+During exploratory data analysis, you will be tempted to write lots of analysis into one big, impressive, start-to-finish script. 
+Though it's fine to write such a script during a long analysis meeting, this practice is error-prone.  
+It subtly encourages poor practices such as not clearing the workspace and not loading fresh data. 
+It's important to take the time to organize scripts in a clean manner and to avoid mistakes.
+
+A well-organized analysis script starts with a completely fresh workspace and explicitly loads data before analyzing it.
+This encourages data manipulation to be done earlier in the workflow (that is, during construction).
+It also and prevents you from accidentally writing pieces of analysis code that depend on one another, leading to the too-familiar ``run this part, then that part, then this part'' process.
```

Comment:
I understand what you mean by "_too-familiar ``run this part, then that part, then this part'' process_" but I do not think we can use this language here. What about:

_It also and prevents you from accidentally writing pieces of analysis code that depend on one another and requires manual instructions  for all required code snippets be run in the right order_

Reply:
Adopted you suggestion in c80a095.